docs/architecture.md

```diff
@@ -0,0 +1,44 @@
+# Architecture
+
+This documents the desired architecture for this project.
+
+### General
+
+The general principle is to arrange the application in a kind of hub-and-spoke structure where each feature is a spoke, and the core application logic serves as the hub. There should ideally be no horizontal dependencies between features, each feature should only rely on the generic functionality provided by the hub. This should be reflected in the directory structure; if a feature directory is deleted, all other features should keep working.
+
+### Root
+
+```
+./
+|- bin          Developer and deployment scripts
+|- docs         Documentation
+|- public       Public assets
+|- src          Client app
+|  |- api             Interface with the Twine HTTP API
+|  |- shared          All code shared between the admin and visitor UIs
```

Comment:
🔥

Reply:
We still need a folder to put common things in. You don't like the way `shared` is structured in the visitor app currently, that's fine, but that stuff still needs to go somewhere.

Reply:
Can you give me an example of some thing that is in our visitor app shared folder that will not fit into another directory.

Reply:
Can you give me an example of some thing that is in our visitor app shared folder that will not fit into another directory.

Reply:
All of the common form components, the nav header. Probably the constants.js file too.

Reply:
`shared` previously made sense as we were splitting between two different sections of the app, the `visitor` view and the `admin` view. I do not think that we need this style of sharing anymore, and any common code can be found under their own directories.

Reply:
What directories? Where do we keep components that are used on several different parts of the app?

Reply:
Can we have a components folder?

Reply:
Can we have a components folder?